examples/qaoa.py

```diff
@@ -1,155 +1,76 @@
-  │           │           │           ZZ──────────ZZ^(-4/13)  │
-┌ │           │           │           │           │           │           ┐
-│ │           │           Rx(0.151π)  ZZ──────────┼───────────ZZ^(-4/13)  │
-│ │           │           │           │           Rx(0.151π)  │           │
-└ │           │           │           │           │           │           ┘
-  ZZ──────────┼───────────ZZ^-0.941   Rx(0.151π)  │           Rx(0.151π)
-┌ │           │           │           │           │           │           ┐
-│ ZZ──────────┼───────────┼───────────┼───────────ZZ^-0.941   │           │
-│ │           ZZ──────────┼───────────ZZ^-0.941   │           │           │
-└ │           │           │           │           │           │           ┘
-  Rx(-0.448π) ZZ──────────┼───────────┼───────────┼───────────ZZ^-0.941
-  │           │           ZZ──────────┼───────────ZZ^-0.941   │
-  │           │           │           │           │           │
-  │           Rx(-0.448π) ZZ──────────┼───────────┼───────────ZZ^-0.941
-  │           │           │           ZZ──────────ZZ^-0.941   │
-┌ │           │           │           │           │           │           ┐
-│ │           │           Rx(-0.448π) ZZ──────────┼───────────ZZ^-0.941   │
-│ │           │           │           │           Rx(-0.448π) │           │
-└ │           │           │           │           │           │           ┘
-  │           │           │           Rx(-0.448π) │           Rx(-0.448π)
-  │           │           │           │           │           │
-  M('m')──────M───────────M───────────M───────────M───────────M
-  │           │           │           │           │           │
-Optimizing objective function ...
-The largest cut value found was 7.
-The largest possible cut has size 7.
-The approximation ratio achieved is 1.0.
-"""
-
+"""Runs the Quantum Approximate Optimization Algorithm on Max-Cut."""
 import itertools
 
 import numpy as np
 import networkx
 import scipy.optimize
+from sympy.parsing.sympy_parser import parse_expr
 
 import cirq
 
 
-def main(repetitions=1000, maxiter=50):
-    # Set problem parameters
-    n = 6
-    p = 2
-
-    # Generate a random 3-regular graph on n nodes
-    graph = networkx.random_regular_graph(3, n)
+def brute_force(graph, n):
+    bitstrings = np.array(list(itertools.product(range(2), repeat=n)))
+    mat = networkx.adjacency_matrix(graph, nodelist=sorted(graph.nodes))
+    vecs = (-1) ** bitstrings
+    vals = 0.5 * np.sum(vecs * (mat @ vecs.T).T, axis=-1)
+    vals = 0.5 * (graph.size() - vals)
+    return max(np.round(vals))
 
-    # Make qubits
-    qubits = cirq.LineQubit.range(n)
 
-    # Print an example circuit
-    betas = np.random.uniform(-np.pi, np.pi, size=p)
-    gammas = np.random.uniform(-np.pi, np.pi, size=p)
-    circuit = qaoa_max_cut_circuit(qubits, betas, gammas, graph)
-    print('Example QAOA circuit:')
-    print(circuit.to_text_diagram(transpose=True))
+def qaoa(booleans, repetitions, maxiter, p):
```

Comment:
I know it wasn't documented in the previous version either, but can we add documentation on what these parameters are?

Reply:
Please address this comment.

Reply:
Sorry about missing this. Added mypy types and comments.